### ISSUE TITLE
[Admin] Unsupported use of can? with array of actions

### ISSUE THREAD

**Issue by spaghetticode** [solidusio/solidus#4433]
**Solidus Version**:
all including master

**Description**
In some views we're probably using `cancancan` in an unsupported way. 

Instead of passing a single action, we provide an array with multiple values. For example, in [app/views/spree/admin/users/orders.html.erb](https://github.com/solidusio/solidus/blob/aa77140e8c5b07ab597126ba76524a3931bbb449/backend/app/views/spree/admin/users/orders.html.erb#L41) we're doing:
```erb
<% if can?([:admin, :edit], order) %>
```
which seems to mean that if the user has permissions for both actions, then they should be able to pass the check. 

Unfortunately, this works only for users that can `:manage` the resource.

Let's take a look at the [cancancan codebase](https://github.com/CanCanCommunity/cancancan/blob/develop/lib/cancan/relevant.rb#L8), where the rule check happens: 

```ruby
    def relevant?(action, subject)
      subject = subject.values.first if subject.class == Hash
      @match_all || (matches_action?(action) && matches_subject?(subject))
    end

    private

    def matches_action?(action)
      @expanded_actions.include?(:manage) || @expanded_actions.include?(action)
    end
```
 
we're mostly interested in the `matches_action?` method, where the first part is verified if the user has `:manage` permissions, while the second is verified when the one dimensional array `@expanded_actions` includes `action`, which is an array in our case... and this will never happen.

I don't know if the array match was dropped at some point in the past or if the feature was never there (the [documentation](https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md) doesn't mention that feature), anyway I think we should change our codebase in order to support the current behavior.

Also, checking `:admin` seems to be a bit redundant, as it's going to [be checked at the controller level](https://github.com/solidusio/solidus/blob/master/backend/app/controllers/spree/admin/base_controller.rb#L29) when checking permissions for the `edit` action.

**Comment by jacobherrington** [comment]
Glancing at the docs it seems like `can` is able to receive an array, but `can?` cannot :smile: 

```
# valid
can [:admin, :manage], Spree::Order, number: 'R987654321'
```

It's not too much of a leap to:

```
# invalid
can?([:create, :update], Spree::ProductProperty)
```

**Comment by adammathys** [comment]
It looks like this was fixed by #4705
